### ISSUE TITLE
Handle messages above the character limit

### ISSUE THREAD

**Issue by hkwu** [hydrabolt/discord.js#653]
There's currently nothing to handle messages which exceed the 2000 character limit.

Ideally the client should:
* split messages over the limit and send them as individual messages
* prefer splits by lines instead of characters to make the messages less awkward to read
* re-wrap code blocks that have been split over messages (preserving the syntax highlighting)

Sample implementation (taken from [here](https://github.com/hydrabolt/discord.js/pull/641)):
```js
const MESSAGE_CHAR_LIMIT = 2000;

const splitString = (string, prepend = '', append = '') => {
  if (string.length <= MESSAGE_CHAR_LIMIT) {
    return [string];
  }

  const splitIndex = string.lastIndexOf('\n', MESSAGE_CHAR_LIMIT - prepend.length - append.length);
  const sliceEnd = splitIndex > 0 ? splitIndex : MESSAGE_CHAR_LIMIT - prepend.length - append.length;
  const rest = splitString(string.slice(sliceEnd), prepend, append);

  return [`${string.slice(0, sliceEnd)}${append}`, `${prepend}${rest[0]}`, ...rest.slice(1)];
};
```

**Comment by macdja38** [comment]
Would be nice if this supported codeblocks as well. eg, when splitting a message that has codeblocks that suround the split point it would add codeblock start and endings at the split point.

**Comment by Gawdl3y** [comment]
Implemented in 7da00cf0fe4ab0cf021235220895216ac96c7a4f. Codeblocks are supported using the `prepend` and `append` options for the `split` message option.
